genki_wave/asyncio_runner.py

```diff
@@ -189,14 +178,15 @@ def _run_asyncio(
         protocol: An object that knows how to process the raw data sent from the Wave ring into a structured format
                   and passes it along between `producer` and `consumer`.
     """
-    # A singleton that sends messages about whether the data transfer has been canceled.
+    # TODO(robert): Catch a keyboard interrupt and gracefully shut down. Non-trivial to implement.
+
+    # A singleton that sends messages about whether the data transfer has been canceled. A slightly hacky fix
+    # to the problem of not handling keyboard interrupts
     comm = CommunicateCancel()
-    loop = get_or_create_event_loop()
-    loop.add_signal_handler(signal.SIGINT, make_sigint_handler(comm))
```

Comment:
I reverted this commit because this is not possible to do unless we are in the main thread. We must find some other way to handle Ctrl-C.